container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -340,10 +353,22 @@ function run_rpm_tempest {
     # List Tempest packages
     rpm -qa | grep tempest
 
+    tempest cleanup --init-saved-state
+
+    if [[ ! ${#TEMPEST_EXTRA_IMAGES_NAME[@]} -eq 0 ]]; then
+        upload_extra_images
+    fi
+
     eval discover-tempest-config ${TEMPESTCONF_ARGS} ${TEMPESTCONF_OVERRIDES} \
     && tempest run ${TEMPEST_ARGS}
     RETURN_VALUE=$?
 
+    # Run tempest cleanup to delete any leftover resources
+    # when not in debug mode
+    if [ ${TEMPEST_DEBUG_MODE} == false && ${TEMPEST_CLEANUP} == true ]; then
+        tempest cleanup
+    fi
```

Comment:
If you want to use `&&` and `[` then the correct syntax is this:
```
[ ${TEMPEST_DEBUG_MODE} = false] && [ ${TEMPEST_CLEANUP} = true ]
```
otherwise you should use `[[`:
```
[[ ${TEMPEST_DEBUG_MODE} == false] && [ ${TEMPEST_CLEANUP} == true ]]
```